### PR TITLE
Add fancy GPS system

### PR DESCRIPTION
**Pull request by merlinND**

### CODE REVIEW

File: ts/src/main/scala/ch/epfl/ts/component/utils/Timekeeper.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// TODO: replace the `parent` argument by simply using `context.parent`?
+class Timekeeper(val parent: ActorRef, val timeBase: Long, val period: FiniteDuration) extends Actor {
+  
+  private val timeOffset: Long = System.currentTimeMillis()
+  
+  private val timer = new Timer()
+  private class SendTheTime extends java.util.TimerTask {
+    def run() {
+      val current = timeBase + (System.currentTimeMillis() - timeOffset)
+      parent ! TheTimeIs(current)
```
Comment (sygi) [inline ts/src/main/scala/ch/epfl/ts/component/utils/Timekeeper.scala:37]: why using parent, if you can use `send`?

File: ts/src/main/scala/ch/epfl/ts/engine/HybridMarketSimulator.scala [hunk 2]
```diff
@@ -14,21 +19,42 @@ import akka.actor.ActorLogging
...
+    }
+    
+    case 'ChangeMarketRules => {
       log.info("Hybrid market: changing rules")
       changeRules
-    case q: Quote if (!isSimulating) =>
+      
+      // We now enter full simulation mode, and we need an actor
+      // to take care of the keeping of the time
+      val timekeeper = context.actorOf(Props(classOf[Timekeeper], self, lastHistoricalTime, timekeeperPeriod), "SimulationTimekeeper")
```
Comment (sygi) [inline ts/src/main/scala/ch/epfl/ts/engine/HybridMarketSimulator.scala:43]: ah, ok, this is fine.

File: ts/src/test/scala/ch/epfl/ts/test/TestHelpers.scala [hunk 3]
```diff
@@ -1,7 +1,7 @@
 package ch.epfl.ts.test
 
-import scala.concurrent.duration.DurationInt
 import scala.language.postfixOps
+import scala.concurrent.duration.DurationInt
```
Comment (sygi) [inline ts/src/test/scala/ch/epfl/ts/test/TestHelpers.scala:4]: sort back :P

Comment (merlinND) [inline ts/src/test/scala/ch/epfl/ts/test/TestHelpers.scala:4]: :no_good:

**Comment by sygi** [comment]
ok, fine, lets merge.

**Comment by merlinND** [comment]
I can haz squirrel?

**Comment by sygi** [comment]
:squirrel:

**Pull request merged**